sqlmesh/core/macros.py

```diff
@@ -349,6 +350,25 @@ def engine_adapter(self) -> EngineAdapter:
             )
         return self.locals["engine_adapter"]
 
+    def _coerce(self, expr: exp.Expression, typ: t.Type) -> t.Any:
+        """Coerces the given expression to the specified type on a best-effort basis."""
+        try:
+            if typ is None:
+                return expr
+            if isinstance(expr, typ):
+                return expr
+            elif issubclass(typ, exp.Expression):
+                return parse_one(expr.sql(self.dialect), into=typ, read=self.dialect)
```

Comment:
here, since you're doing an imediate parse_one from the sql, you don't need to specify the dialect.

you can just do parse_one(expr.sql(), into=typ)